farmdata2_modules/fd2_tabs/fd2_config/config.html

```diff
@@ -1,41 +1,131 @@
+            Configuration Saved
+        </div>
+        <p data-cy='intro'>This configuration page contains setting for FarmData2 specific settings.</p>
+        <div>
+            <label>Labor Data: 
+                <dropdown-with-all data-cy="labor-dropdown" :selected-val="selectedLaborConfig" :dropdown-list="laborConfigs" @selection-changed='setNewLabor'></dropdown-with-all>
+            </label>
+            <button type="button" class="btn" style="background-color:transparent" data-toggle="tooltip" data-placement="right" title="Set the labor data input to required, optional, or hidden (optional and not displayed).">
+                <span class="glyphicon glyphicon-question-sign">
+                </span>
+            </button>
+        </div>
+        <button data-cy="save-button" class="btn center-block" style='background-color: #da4f3f; color: white; font-size: large;' :disabled='configNotReady' @click='saveConfig'>Save</button>
+    </div>
 
     <script>
-      function getIt() {
-        getConfiguration().then((response) => {
-            console.log(response.data)
-        })
-      }
-
-      function makeOptional() {
-        let config={
-          "labor": "Optional",
-        }
-
-        getSessionToken().then((token) => {
-          setConfiguration(config, token)
-        })
-      }
-
-      function makeRequired() {
-        let config={
-          "labor": "Required",
-        }
+        new Vue({
+            el: '#config',
+            components: {
+                'dropdown-with-all': DropdownWithAllComponent,
+            },
+            data: {
+                sessionToken: null,
+                configObj: {},
+                selectedLaborConfig: '',
+                laborConfigs: ['Required', 'Optional', "Hidden"],
+                isSaved: false,
+                errBanner: null,
+            },
+            methods: {
+                setNewLabor(newLaborConfig) {
+                    this.selectedLaborConfig = newLaborConfig
+                },
+                saveConfig() {
+                    this.configObj.labor = this.selectedLaborConfig
+                    setConfiguration(this.configObj, this.sessionToken) 
+                        .then(() => {
+                            this.isSaved = true
+                            setTimeout(() => { 
+                                this.isSaved= false
+                            }, 3000)
 
-        getSessionToken().then((token) => {
-          setConfiguration(config, token)
-        })
-      }
+                        }) 
+                        .catch((err) => { 
+                            if (err.response) {
+                                // The request was made and the server responded with a status code
+                                // that falls out of the range of 2xx
+                                this.errBanner = true
+                                console.log('Error code that falls out of 2xx.')
```

Comment:
These console.log messages are not necessary. If there is a persistent error the dev tools can provide the necessary information.